tool/tsh/tsh.go

```diff
@@ -1784,6 +1788,52 @@ func makeClient(cf *CLIConf, useProfileLogin bool) (*client.TeleportClient, erro
 	return tc, nil
 }
 
+// The order of default proxy ports to try, in order that they will be tried.
+var defaultWebProxyPorts = []int{
+	defaults.HTTPListenPort, defaults.StandardHTTPSListenPort,
+}
+
+// setClientWebProxyAddr configures the client WebProxyAddr and SSHProxyAddr
+// configuration values. Values that are not fully specified via configuration
+// or command-line options will be deduced if necessary.
+//
+// If successful, setClientWebProxyAddr will modify the client Config in-place.
+func setClientWebProxyAddr(cf *CLIConf, c *client.Config) error {
+	// If the user has specified a proxy on the command line, and one has not
+	// already been specified from configuration...
+
+	// NB: I think this *might* be the origin of the behaviour I described in
+	//     this conversation:
+	// https://gravitational.slack.com/archives/C0DF0TPMY/p1617251076408700
```

Comment:
I don't think we should be including links to our private Slack conversations in the open-source code.

Reply:
Deleted